Assets/ZSSRichTextEditor.js

```diff
@@ -346,6 +329,29 @@ ZSSEditor.getYCaretInfo = function() {
     return this.caretInfo;
 };
 
+/**
+ *  @brief      Whenever this method is called, a check will be performed on the caret position
+ *              to figure out if it needs to be wrapped in a paragraph node.
+ */
```

Comment:
Done!  And more!  Read below...

This was actually very enlightening and an awesome example of why documentation is important.  After reviewing the comments and seeing the code I realized something was wrong.  I ended up:

1. Moving the code to the field class.
2. Adding some controls that were missing from this method, and that where outside of it when they shouldn't have been.

While fixing this I also found and fixed a related issue with the title field, that was showing some strange characters when trying to delete or modify it.